### ISSUE TITLE
Failure during second aomenc pass with two different video files.

### ISSUE THREAD

**Issue by DillFrescott** [staxrip/staxrip#488]
**Describe the bug**
Error trying to encode the movie Ordinal Scale into AV1 via aomenc.

**Expected behavior**
I expect it to not have a critical error.

**How to reproduce the issue**

Default aomenc encoding settings, except i changed the cpu speed to 5 and the crf to 30. I tried two different versions of the movie and both failed during the second pass.

**Provide information**

https://gist.github.com/DillFrescott/28cb3712a1760bbbb8a97bf74b593728

**Additional context**
My cpu should be under the gist but its a brand new 5800x so it should work fine.

**Comment by Dendraspis** [comment]
This looks like a memory issue. The 2nd pass runs about 139min till the problem occurs, so I bet there was not enough memory left for aomenc.

How much GB of RAM do you have?
Were other, maybe big, apps running meanwhile?

**Comment by DillFrescott** [comment]
I have 32 GB at 3200 mhz. I have a little meter that monitors performance and the ram never went above about half usage. I had discord and Minecraft open at the same time during the second test. The first time it failed i believe only discord was open.

**Comment by DillFrescott** [comment]
Would running staxRip as admin help at all?

**Comment by Dendraspis** [comment]
Interesting, but Minecraft should be off during testing.
 Something seems to interrupt the 2nd pass.

> Would running staxRip as admin help at all?

l doubt that.  :thinkig:

**Comment by DillFrescott** [comment]
Does the error code -1073741819 (0xC0000005) say what it is exactly? Or just a memory issue?

**Comment by DillFrescott** [comment]
Keep in mind it worked fine for another movie that was just about the same length. Could it be the movie itself?

**Comment by DillFrescott** [comment]
I tried one ~14 GB version of the movie and the second version was ~25 GB. Both failed but I never copied the error log from the first time it failed.

**Comment by DillFrescott** [comment]
Ok I updated aomenc to the newest build. Hope it fixes it

**Comment by DillFrescott** [comment]
Ok. Updated aomenc and ffmpeg. It still threw an error. Since its most likely a memory issue I am going to try something I probably should have done at the start and reboot my computer. It probably won't fix it but its worth a try.

I'll let it run overnight and report back in the morning.

**Comment by Dendraspis** [comment]
Have you also tested encoding a complete different source file?
Maybe the source is broken?!

**Comment by stax76** [comment]
It could be access right related.

```
Development> .\Err.exe 0xC0000005
# for hex 0xc0000005 / decimal -1073741819
  ISCSI_ERR_SETUP_NETWORK_NODE                                   iscsilog.h
# Failed to setup initiator portal. Error status is given in
# the dump data.
  STATUS_ACCESS_VIOLATION                                        ntstatus.h
# The instruction at 0x%p referenced memory at 0x%p. The
# memory could not be %s.
  USBD_STATUS_DEV_NOT_RESPONDING                                 usb.h
# as an HRESULT: Severity: FAILURE (1), FACILITY_NONE (0x0), Code 0x5
# for hex 0x5 / decimal 5
  WINBIO_FP_TOO_FAST                                             winbio_err.h
# Move your finger more slowly on the fingerprint reader.
# as an HRESULT: Severity: FAILURE (1), FACILITY_NULL (0x0), Code 0x5
  ERROR_ACCESS_DENIED                                            winerror.h
# Access is denied.
# 5 matches found for "0xC0000005"
```

**Comment by 44vince44** [comment]
Would this imply the file is read over a network ???

**Comment by DillFrescott** [comment]
> 
> 
> Have you also tested encoding a complete different source file?
> Maybe the source is broken?!

yes i have

**Comment by DillFrescott** [comment]
> 
> 
> Would this imply the file is read over a network ???

the file is on my SSD along with staxrip. No network transfer happening here...

**Comment by DillFrescott** [comment]
It failed again. Here is the new log.

https://gist.github.com/DillFrescott/290847fa54ffc41aaaf848b5fd05103e

**Comment by stax76** [comment]
It's the same error code 0xC0000005, maybe Process Monitor can be used to get more error info:

See this post:

https://forum.doom9.org/showthread.php?p=1932802#post1932802

But I've never used Process Monitor for debugging, I cannot tell you how to do it, you can try to google how to use Process Monitor to get extended error info.

**Comment by DillFrescott** [comment]
I've never attempted something like that before. But I can give it a try in a bit I guess

**Comment by DillFrescott** [comment]
I'm sorry I don't know how to do it

**Comment by DillFrescott** [comment]
Im sorry, I have no idea what I'm doing with Process Monitor. I just want to convert this movie to AV1

**Comment by DillFrescott** [comment]
I'm going to try totally reinstalling staxrip and trying a third, totally different version of the same movie.

**Comment by Dendraspis** [comment]
If possible, try starting StaxRip from another Partition like `D:\`. Also try disabling any AntiVirus software.  :thinking:

**Comment by DillFrescott** [comment]
Sure. I can try that. Now that you mention it, I do have bitdefender free antivirus installed. I'm going to totally uninstall it and try again. If that fixes it at least we know it has a good chance of that being the issue.

**Comment by DillFrescott** [comment]
I can also try on a totally separate machine if all else fails.

**Comment by DillFrescott** [comment]
I'm getting this error now. It seems totally different.

https://gist.github.com/DillFrescott/c5b9c5c949c0d3d793bb04ca225dbf4f

**Comment by stax76** [comment]
In Process Monitor go to:

Filter > Filter

Define following conditions:

Process Name Is aomenc.exe then include
Process Name Is ffmpeg.exe then include

Click _Add_
Click _OK_

Edit > Clear Display

Start your StaxRip Job

After the error occurs:

File > Save

That wasn't so difficult.

Upload the file.

> Apparently it doesn't like being compressed into an archive and moved back...

I don't understand what you mean with that.

**Comment by Dendraspis** [comment]
> I even specified a width and height and it still threw a different error. I fixed it by renaming the main folder staxrip is in.

I would like to know what this means in detail. Because just renaming a folder would change nothing, except from starting with new settings. 🤔

**Comment by DillFrescott** [comment]
The settings stay because I select it to use the same folder for the settings

**Comment by DillFrescott** [comment]
> In Process Monitor go to:
> 
> 
> 
> Filter > Filter
> 
> 
> 
> Define following conditions:
> 
> 
> 
> Process Name Is aomenc.exe then include
> 
> Process Name Is ffmpeg.exe then include
> 
> 
> 
> Click _Add_
> 
> Click _OK_
> 
> 
> 
> Edit > Clear Display
> 
> 
> 
> Start your StaxRip Job.
> 
> 
> 
> After the error occurs:
> 
> 
> 
> File > Save
> 
> 
> 
> That wasn't so difficult.
> 
> 
> 
> Upload the file.
> 
> 
> 
> > Apparently it doesn't like being compressed into an archive and moved back...
> 
> 
> 
> I don't understand what you mean with that.

Looks like you seem to understand the program a bit better than I did.

**Comment by DillFrescott** [comment]
Ok. I ram it on a separate drive and set the output path to just the same as the original source and it seems to be working. Would you still like me to move it back onto the C: drive and hook the process monitor up to it?

**Comment by Dendraspis** [comment]
It seems to be obvious, that it is/was an access right issue. So we found the problem and you know how to solve it. I would say no more testing needed.  😊

**Comment by DillFrescott** [comment]
Ok. Just wanted to ask just in case you guys wanted to know the exact details. Thanks for the help!

**Comment by DillFrescott** [comment]
Ok. Apparently its related to the source file. I tried the old configuration on the main SSD and it worked fine with a totally different episode of a show...

**Comment by DillFrescott** [comment]
There was a version of ordinal scale that actually worked fine on the other drive so I'm assuming it had something to do with those first two versions of the movie...

**Comment by DillFrescott** [comment]
Ok. It happened with another movie. I am currently hooking aomenc and ffmpeg up to process monitor as I write this and will have a dump in probably about 5 hours roughly, or whenever it decides to throw the error.

**Comment by DillFrescott** [comment]
Ok. The crash happened again, and of course process monitor is not responding. It wouldn't even open the window back up from being minimized. I guess when aomenc crashed it took process monitor with it? Or is it just a fluke?

Not sure what to do...

**Comment by 44vince44** [comment]
It seems you have a hardware problem there....

**Comment by DillFrescott** [comment]
Ah, hope not. I'll try a different machine.

**Comment by 44vince44** [comment]
Of course, try a different machine. This bug does not seem not be related to the encoder or to staxrip.

> I'd say it's rather windows being windows
We're all using Windows, and we have no such problems. I don't like to defend Microsoft, but I must admit that Windows is becoming very robust. It is not Windows being Windows, it's something else here.

**Comment by DillFrescott** [comment]
Hmmm. Everything else works fine. I'm thinking process monitor not working right could have been a glitch of some sort.

I noticed that it was also difficult to click on the start menu so my best guess is that it's because I hadn't restarted the machine in a while. Other than that every component is brand new and working fine. I can give you the full specs of the machine.

**Comment by 44vince44** [comment]
>  I hadn't restarted the machine in a while

OMG. Before opening an issue, and for the 5 days that have passed, hasn't it occur to you to restart ?
🤣

**Comment by DillFrescott** [comment]
Lol I tried that at the first part of the issue but I forget to check simple things sometimes lmao

**Comment by DillFrescott** [comment]
I'm going to run it again with process monitor after restarting it when I get home and see what happens.

**Comment by 44vince44** [comment]
@DillFrescott This is obviously a problem you have with your OS or hardware... please close this because there is not bug!

**Comment by DillFrescott** [comment]
Ok. But if it happens on another machine I am going to reopen it.

Thanks for the help btw

**Comment by DillFrescott** [comment]
@44vince44 

Actually I'm going to reopen it.

I am still going to run process monitor and send the results of that. There's a possibility it's not hardware/os related and has to do with the video file itself.

I thought about it and the fact that it only happens with certain files leads me to believe there might be an actual bug here.

**Comment by 44vince44** [comment]
🤣

**Comment by Dendraspis** [comment]
> OMG. Before opening an issue, and for the 5 days that have passed, hasn't it occur to you to restart ?

I shouldn't tell you my uptimes. 🤣

> But if it happens on another machine I am going to reopen it.

👍🏻

**Comment by DillFrescott** [comment]
@44vince44 If it is hardware related at all, I would think its related to the new zen 3 ryzen cpu. Which is the one i have (5800x)

Has anyone confirmed that this new cpu architecture no issues with the software?

**Comment by 44vince44** [comment]
You understand that this thread in intended to debug Staxrip and to some point the tools included.

 Please understand this by reading and processing the data, not sending to garbage immediately, just process some of it, ok ?

1) This is no problem with STAXRIP which is a frontend for the aomenc tool, if staxrip was the cause, the crash wouldn't happen in the middle of a processing stage
2) You must do a test on the same file TWICE and see if the crash happens at the same time index. If it does, there is a problem with the aomenc tool. If not, the problem is external to aomenc

OK?

**Comment by DillFrescott** [comment]
> 
> 
> You understand that this thread in intended to debug Staxrip and to some point the tools included.
> 
> Please understand this by reading and processing the data, not sending to garbage immediately, just process some of it, ok ?
> 
>     1. This is no problem with STAXRIP which is a frontend for the aomenc tool, if staxrip was the cause, the crash wouldn't happen in the middle of a processing stage
> 
>     2. You must do a test on the same file TWICE and see if the crash happens at the same time index. If it does, there is a problem with the aomenc tool or the video. If not, the problem is external to aomenc
> 
>     3. If first pass is successful, it is likely that the file has no problem
> 
> 
> OK?

Noted. Not sure what you mean by "sending to garbage immediately" though.

I would consider this a staxrip issue, since I am trying to encode a video using starxrip and it isn't working...

**Comment by 44vince44** [comment]
@DillFrescott

> @44vince44 If it is hardware related at all, I would think its related to the new zen 3 ryzen cpu.
Nope it would be more memory overheating or file access problem due to HDD, or chipset problem due to faulty motherboard. But not the Ryzen itself.

It could also be software related when the OS has a problem.

> Not sure what you mean by "sending to garbage immediately" though.

The answer is here:
"I would consider this a staxrip issue, since I am trying to encode a video using starxrip and it isn't working..."

meaning you have sent to garbage this, what I told you before:
"...STAXRIP which is a frontend for the aomenc tool, if staxrip was the cause, the crash wouldn't happen in the middle of a processing stage"

**Comment by 44vince44** [comment]
Now please do the test `2)You must do a test on the same file TWICE and see if the crash happens at the same time index. If it does, there is a problem with the aomenc tool or the video. If not, the problem is external to aomenc` because this will answer the question.

**Comment by DillFrescott** [comment]
Aomenc comes bundled with staxrip. It seems like part of the program since there is a dedicated section for it within staxrip.

This is much like a software that runs a web service not working, and the developers blaming it on nginx because that's what it uses as its backend.

**Comment by JJKylee** [comment]
Well... You are so wrong about it.

**Comment by DillFrescott** [comment]
@JJKylee Please explain, because obviously I don't understand it correctly.

**Comment by JJKylee** [comment]
Do all FFmpeg frontend developers have to take responsibility if FFmpeg has a problem?
If the problem occurs from within StaxRip, that's the issue of StaxRip.
But if the problem occurs even without it, it's not its fault.

You can run aomenc _without_ StaxRip. Why don't you run it on CMD and see if the problem still occurs?
If you find that the problem occurs _only when aoemenc is used via StaxRip_, then you can say it's StaxRip's problem.

**Comment by Dendraspis** [comment]
StaxRip is no equivalent to a car or a kitchen, were the builder is the seller and also responsive for the whole product.
StaxRip is just a tool to work with all these tools for better and faster workflow. 

> Aomenc comes bundled with staxrip. It seems like part of the program since there is a dedicated section for it within staxrip.

If it works better for you, you can also get just the exe file and collect everything else by yourself.  😁

**Comment by DillFrescott** [comment]
Ok. I admit I was wrong. What you say is correct. The way I thought about it was more of it being a problem with how staxrip interacts with aomenc causing the issue.

**Comment by DillFrescott** [comment]
Would it be best to just keep the issue closed until I have a full diagnostic report to send over?

**Comment by 44vince44** [comment]
The thing you must understand is here:

"...STAXRIP which is a frontend for the aomenc tool, if staxrip was the cause, the crash wouldn't happen in the middle of a processing stage"

This DOESN't mean I'm throwing the blame. I am just telling you that Staxrip, as a frontend, creates and sends commands. Either the command is correct or not. If it's not then the crash happens instantly when the command is sent. However, if the command begins processing, and fails some hours later, then it is likely the frontend is OK and the problem is somewhere else.

So this is why I am asking you to do the same test TWICE, on the same file, and show us the logs.
- if the crash happens at the same time index then this is either Staxrip/aomenc/video file issue
- if the crash happens at different time indices then this is a problem with your OS or you hardware.

**Comment by DillFrescott** [comment]
@44vince44 ok. i understand now. It was my bad, I misinterpreted what you were saying.

I'll run it twice on the same file like you said.

**Comment by Dendraspis** [comment]
> You can run aomenc _without_ StaxRip. Why don't you run it on CMD and see if the problem still occurs?
> If you find that the problem occurs _only when aoemenc is used via StaxRip_, then you can say it's StaxRip's problem.

This is a very good advice. 

> I could discuss it with the aomenc developers as well.

First you should try running aomenc from console by hand to verify the problem also occurs without StaxRip. The *position* of the encoding process is also very important to see, if it's everytime the same position/frame.

**Comment by 44vince44** [comment]
@DillFrescott I'll ask you, in addition to the logs, to check after each crash, the size of the video stream generated in the temp folder. This information won't appear in the log, It could be important!

**Comment by DillFrescott** [comment]
@44vince44 I'll be sure to save both temp folders for both tests

**Comment by 44vince44** [comment]
Great, be sure to use the exact same settings in both tests! Can't wait to see the result!

**Comment by DillFrescott** [comment]
One think I do know is that it either fails right at the end of the second pass or at about 90% of the way. So of course it has to fail at the very end and not the beginning so I have to wait LOL

**Comment by Dendraspis** [comment]
..... ***TIP:*** You can copy the whole commandline argument into the clipboard:

<img src="https://user-images.githubusercontent.com/50659978/107161861-c9208400-699f-11eb-8b2a-6c86c7ec633f.PNG">

**Comment by DillFrescott** [comment]
@Dendraspis Staxrip seems to break the video up into parts before it begins the aomenc pass.

This is the command line:

aomenc v2.0.0-1217-gc1047c799

C:\Users\cross\Desktop\staxrip1111\Apps\Encoders\ffmpeg\ffmpeg.exe -i "C:\Users\cross\Desktop\encoding\weathering with you_temp\weathering with you_av1.avs" -f yuv4mpegpipe -strict -1 -loglevel fatal -hide_banner - | C:\Users\cross\Desktop\staxrip1111\Apps\Encoders\aomenc\aomenc.exe - --disable-warning-prompt --passes=2 --pass=1 --cq-level=30 --threads=32 --bit-depth=10 --end-usage=q --cpu-used=5 --auto-alt-ref=1 --static-thresh=0 --tile-columns=2 --tile-rows=1 --enable-tpl-model=1 --coeff-cost-upd-freq=2 --mode-cost-upd-freq=2 --mv-cost-upd-freq=2 --fpf="C:\Users\cross\Desktop\encoding\weathering with you_temp\weathering with you_av1.txt" -o "C:\Users\cross\Desktop\encoding\weathering with you_temp\weathering with you_av1_out.webm"


Would I need to process it in a similar way with just the aomenc command line? This is getting a bit out of my area of knowledge. I apologize.

**Comment by DillFrescott** [comment]
what is the --fpf= ? and why is it a text file?

**Comment by JJKylee** [comment]
> One think I do know is that it either fails right at the end of the second pass or at about 90% of the way. So of course it has to fail at the very end and not the beginning so I have to wait LOL

If it's true, then I'm definitely sure it has _nothing to do with StaxRip_ cuz passing on the command to aomenc is all there is that StaxRip does.
Either aomenc, or many other problems including hardware as well.

**Comment by DillFrescott** [comment]
@JJKylee ok. Even if it has nothing to do with staxrip i appreciate you guys still willing to help me even though you dont even have to help whatsoever at all. I truly appreciate it. <3

**Comment by DillFrescott** [comment]
These issues I'm having are very frustrating so having people to help work out what exactly is going on is a lifesaver. :)

**Comment by 44vince44** [comment]
@JJKylee @Dendraspis it is almost certain that testing the command directly will not help. Because we know the crash will happen.

@DillFrescott the command syntax you see is a composite command of ffmpeg piping the avs script to the aomenc encoder.
You'll have to do both passes, so let Staxrip do it for you. Don't do the cli yourself, do the test I told you...

**Comment by DillFrescott** [comment]
@44vince44 ok. It's in progress right now.

**Comment by 44vince44** [comment]
Great, check your global memory usage. Now first pass is fast. The important thing is to check when the second pass begins, the memory usage should increase then stabilize.

**Comment by DillFrescott** [comment]
Yes. First pass is nearly done with 15.75 GB used total by the system.

Now the second pass is beginning. We are at 16.35 GB used with 15.xx GB free.

**Comment by DillFrescott** [comment]
It's hovering right around 16 GB used at the moment. Which seems normal enough

**Comment by Dendraspis** [comment]
@DillFrescott 
Why do you use ***ffmpeg*** as decoder??  😲  Please try the default  ***avs2pipemod*** instead!

> Staxrip seems to break the video up into parts before it begins the aomenc pass.

Absolutely not! You set up a 2-pass encode, so you have to start the first pass as shown above and then the second pass with `--pass=2` instead.

**Comment by DillFrescott** [comment]
@Dendraspis oops. I didn't even know! After these two tests I can switch it to using the avs2pipemod. I am not terribly good with this stuff, just enough to do basic encodes lol.

**Comment by 44vince44** [comment]
@Dendraspis the default pipe for aomenc is ffmpeg ... i've just tested it on a frech install

**Comment by JJKylee** [comment]
I've seen that too.
I haven't used aomenc for a while, so I haven't found out that there's **no** Decoder-Pipe option in aomenc. _Only ffmpeg piping is allowed_. Please confirm this.
Maybe this is another issue.
Version 2.1.7.8.

**Comment by JJKylee** [comment]
But @DillFrescott, the above mentioned two issues are **_not_** related to your problem at all.
Thanks to you, we could find two issues in StaxRip aomenc command composition window. 😉

**Comment by Dendraspis** [comment]
Confirmed.  Didn't realize it while testing, too! 😲  🤢 

@DillFrescott 
So sorry, there is no option for that for ***aomenc*** at the moment.   😞 

But we could try using ***avs2pipemod*** if the CMD call also fails.  🤔

**Comment by 44vince44** [comment]
@DillFrescott check memory usage again a bit later!

**Comment by DillFrescott** [comment]
@Dendraspis perfectly fine. Aside from it crashing, the process right now delivers excellent results for the final video, both in size and visual quality!

**Comment by Dendraspis** [comment]
@44vince44 

> I should open an issue fort that ?

You have opened enough issues for the whole year.  🤣   But feel free. Maybe it will be concerned in 2 years or so.  😁

**Comment by DillFrescott** [comment]
@44vince44 It's at ~17 GB used with ~14.8 gb free

**Comment by 44vince44** [comment]
now that could be a problem if it's still increasing...

**Comment by DillFrescott** [comment]
I'll keep an eye on it

**Comment by 44vince44** [comment]
@JJKylee @Dendraspis I've checked an older versions of Staxrip, v2.1.4.2, the pipe is ffmpeg for aomenc.

**Comment by DillFrescott** [comment]
@44vince44 I closed every unnecessary background process and it brought it down a bit but it is indeed rising albeit very slowly.

**Comment by 44vince44** [comment]
That looks like a memory leak. This could be the issue. What is the size of your video ? 1920x1080 or more ?

**Comment by Dendraspis** [comment]
@44vince44 
Of course it hadn't changed. Seems to be the most basic pipe option in early days of aomenc just to make it work.

@DillFrescott 
That could be the problem - normaly the used RAM should stop at some point.

**Comment by DillFrescott** [comment]
The only things open at the moment are firefox, process monitor, and staxrip btw. It seems to have stabilized actually. around just under 16 GB.

The video size is 1920x1080 yes.

**Comment by 44vince44** [comment]
@Dendraspis changing to avs2pipe / vspipe would improve performance ?

**Comment by 44vince44** [comment]
> The only things open at the moment are firefox, process monitor, and staxrip btw. It seems to have stabilized actually. around just under 16 GB.

that's good news

**Comment by DillFrescott** [comment]
Ok. It hung around 16 GB for a bit, and now its has started climbing again. It just broke 17 GB.

I can't tell if its aomenc/staxrip causing it though, but I assume it could be since its one of the only things open.

**Comment by DillFrescott** [comment]
Would it most likely be aomenc doing this?

**Comment by 44vince44** [comment]
You can check in the task manager, the details tab, check the process aomenc.exe, it has its own memory usage in the column Memory. There you can see the memory used by the process. just note the approximate value and check in half an hour. it should be similar...

**Comment by Dendraspis** [comment]
> @Dendraspis changing to avs2pipe / vspipe would improve performance ?

Primarily I don't like ffmpeg.  😜   Don't know if it would cause a performance gain. But I feel a lot better when ffmpeg is not involved. 🙄

**Comment by Dendraspis** [comment]
***aomenc*** or ***ffmpeg*** .... I'm curious.

**Comment by 44vince44** [comment]
> I feel a lot better when ffmpeg is not involved. 

I couldn't agree more !!

**Comment by DillFrescott** [comment]
600-630 MB. Guess its something else...

Firefox is using almost 900 MB

ffmpeg is using 260 MB

Process monitor is using 130 MB. 

So I'm not sure whats sucking up so much ram.

**Comment by DillFrescott** [comment]
Honestly think its a bunch of little things taking up so much just spread out.

**Comment by 44vince44** [comment]
you can sort that memory column, so the highest appears on top. it should be aomenc.... right ???

**Comment by DillFrescott** [comment]
It goes

Firefox > aomenc > ffmpeg > process monitor >antimalware service executable > secure system > etc...

**Comment by 44vince44** [comment]
@Dendraspis you have a ryzen as well, how mush is your global memory usage when the system is idle ? or almost

**Comment by DillFrescott** [comment]
secure system is only using 72 MB so everything below that is using less

**Comment by DillFrescott** [comment]
It seems to be stabilizing around 17.8 GB again. Ill wait and see if it continues to climb again.

**Comment by 44vince44** [comment]
> If it has to do with the new ryzen cpu it could be a bios issue. I had to update the bios initially to get the computer to even boot.

That is not surprising, although rare, BIOS updates allows sometimes new processors ids and more memory compatibility. But once the system boots and is working, there shouldn't be unexplainable memory increases...

**Comment by 44vince44** [comment]
The global memory usage that you get on the Performance page, is a commited memory value, it doesn't really reflect the real need of the processes. it is well above. It depends on the available memory and the virtual memory. So If you're still around 17GB I'm almost sure it's not a memory leak, or a memory drain...

**Comment by 44vince44** [comment]
e.g: 
-systems with 64GB tend to idle around 15 GB of committed memory.
-systems with 16GB tend to idle aroung 5GB of committed memory.

So what you are describing looks actually very OK. Your memory usage will not get above 18 GB (if you don't run stuff)

**Comment by Dendraspis** [comment]
> @Dendraspis you have a ryzen as well, how mush is your global memory usage when the system is idle ? or almost

Never saw it in such a state.  😁 

@44vince44  
Did you forget where the ***Edit*** button is?  😉 

---------------

> Firefox > aomenc > ffmpeg > process monitor >antimalware service executable > secure system > etc...

But how many GB did aomenc use ?!?  🤔   But less than 9GB seems to be okay, though.

**Comment by DillFrescott** [comment]
@Dendraspis its currently abetween 600-630 MB

**Comment by 44vince44** [comment]
> Never saw it in such a state. 😁

🤣 🤣 

His aomenc is using around 600 MB

**Comment by 44vince44** [comment]
I'm going to sleep, good night all. I'll check here in the morning, Good night all!

**Comment by Dendraspis** [comment]
Don't get me wrong, but 600MB seems to be too little. 🤔

I think I lost the connection between those hundreds of comments and edits. Can we just summerize:

1. Encoding via StaxRip still produces errors
2. The moment/position/frame is always different
3. Direct execution via cmd was not tested yet

**Comment by DillFrescott** [comment]
@Dendraspis not quite sure about the second one. It could be the same spot each time. The first test is currently running but it takes hours each time so I'll have the results by some time tomorrow if the software cooperates.

**Comment by 44vince44** [comment]
@Dendraspis , I am performing a test here, on my i7, with a video 1920x1080p encoding with aomenc:
1st pass => memory usage of aomenc.exe approximates 350MB
2nd pass => memory usage of aomenc.exe approximates 800MB

This seems reasonable since, in comparison, x265 uses approximately 1.1GB memory in similar conditions!

**Comment by Dendraspis** [comment]
@44vince44 
Thanks, couldn't check it by myself yet.  ☺️

**Comment by DillFrescott** [comment]
After running the test again, it did indeed crash, but so did process monitor again.

Aside from testing it on another machine, there's not much more I can do.

We can just say it's an issue with my computer I guess, and just close the issue for now.

I really wish it would work but this is the second time process monitor has completely stopped responding. :(

**Comment by 44vince44** [comment]
What about doing the same test again to check if the crash happens at the same time index ?

**Comment by DillFrescott** [comment]
I can still do that.

**Comment by 44vince44** [comment]
can you already post the log of the first test please ? and give us the size of the temp file [your video name].webm
You can probably watch that video on mpv.net (it will sure play video only, but it'll give you the time index of the film where the crash happened)

**Comment by Dendraspis** [comment]
If the problem occurs after about 30s after the second pass has started, it has to be a system stability issue.  🤔

**Comment by stax76** [comment]
First I thought maybe file access, but if it's a stability issue then it could be memory access.

```
Development> .\Err.exe 0xC0000005
# for hex 0xc0000005 / decimal -1073741819
  ISCSI_ERR_SETUP_NETWORK_NODE                                   iscsilog.h
# Failed to setup initiator portal. Error status is given in
# the dump data.
  STATUS_ACCESS_VIOLATION                                        ntstatus.h
# The instruction at 0x%p referenced memory at 0x%p. The
# memory could not be %s.
  USBD_STATUS_DEV_NOT_RESPONDING                                 usb.h
# as an HRESULT: Severity: FAILURE (1), FACILITY_NONE (0x0), Code 0x5
# for hex 0x5 / decimal 5
  WINBIO_FP_TOO_FAST                                             winbio_err.h
# Move your finger more slowly on the fingerprint reader.
# as an HRESULT: Severity: FAILURE (1), FACILITY_NULL (0x0), Code 0x5
  ERROR_ACCESS_DENIED                                            winerror.h
# Access is denied.
# 5 matches found for "0xC0000005"
```

**Comment by DillFrescott** [comment]
Sorry, I've been busy all day.

The size of the temp file is 982,500,644 bytes.

I never copied the crash log itself, is it saved somewhere? Sorry about that.

I can play the webm file but the time line is stick to the end the entire time using VLC player. I tried that mpv.net but it seems the domain is for sale.

**Comment by DillFrescott** [comment]
Ah, found it. 

https://gist.github.com/DillFrescott/1a125055b562763a97f586112742e3ef

**Comment by 44vince44** [comment]
>  I tried that mpv.net but it seems the domain is for sale.

@DillFrescott mpv.net is not a website, it is @stax76 's version of the mpv player and it is included in Staxrip. you can open this player from staxrip form Staxrip's menu APPS > Players > mpv.net.

Once mpv.net is open, simply drag and drop your webm file inside it. The file will play and you'll be able to seek inside it.
How to know at what time index the crash happened ? : just pause the player, and seek to the end with the left right arrows. with the comma the point "," "." of your keyboard you can seek frame by frame back and forward. you can make mpv.net display the time index with milliseconds precision by clicking on the time counters.

This is very valuable for our debugging. We can even run later a quick test to check that there is nothing wrong with the video itself, by selecting a range around the crash index and encoding only that range. It would take some minutes and give us an answer.

**Comment by 44vince44** [comment]
@Dendraspis @stax76 I may have some idea what's happening here: our friend is using here Staxrip 2.1.7.0 with tools OTHER than the ones we selected:
- aomenc : he uses a version called "aomenc v2.0.0-1220-g7926ca6cd", but Staxrip 2.1.7.0 is bundled with aomenc 2.0.1-1118-gbe4ee75fd-x64-msvc1929 Patman and i remember that there is some specific reason why Stax76 amd Patman86 chose msvc build.
- ffmpeg : he uses ffmpeg 4.3.2-2021-02-02, but staxrip 2.1.7.0 comes with ffmpeg N-100448-gab6a56773f-x64-gcc10.2.0 Patman.

@DillFrescott hasn't it occured to you that using tools from other sources that ours can be the problem here?

**Comment by DillFrescott** [comment]
The crash happened with the bundled versions as well. I figured that since it happened with those versions it wouldn't matter much.

It was the exact same error as well in the crash log.

Would you like me to do it again with the stock versions?

**Comment by DillFrescott** [comment]
Second test is complete.

The size of the first webm is 982,500,644 bytes

The size of the second test webm is 982,500,644 bytes

I can do the tests with the stock versions of aomenc and ffmpeg as well.

**Comment by Dendraspis** [comment]
That would indicate a broken or out-of-specification source for me.  🤔

**Comment by DillFrescott** [comment]
@Dendraspis Very odd. Def could be the source file.

**Comment by 44vince44** [comment]
@DillFrescott I gave you the solution, and you seem to ignore it:

> @DillFrescott mpv.net is not a website, it is @stax76 's version of the mpv player and it is included in Staxrip. you can open this player from Staxrip's menu APPS > Players > mpv.net.
> 
> Once mpv.net is open, simply drag and drop your webm file inside it. The file will play and you'll be able to seek inside it.
> How to know at what time index the crash happened ? : just pause the player, and seek to the end with the left right arrows. with the comma the point "," "." of your keyboard you can seek frame by frame back and forward. you can make mpv.net display the time index with milliseconds precision by clicking on the time counters.
> 
> This is very valuable for our debugging. We can even run later a quick test to check that there is nothing wrong with the video itself, by selecting a range around the crash index and encoding only that range. It would take some minutes and give us an answer.

**Comment by DillFrescott** [comment]
@44vince44 I'm sorry Vince I've got a lot going on at the moment. I will follow your instructions as soon as I get the chance.

**Comment by Dendraspis** [comment]
Maybe it could help playing the video right before the specific timestamp and see if the player stops, shutters, etc

But for me it's definitely the source file .. after 129 comments. 😂😵😂


@44vince44 
Have a look at discussions. 🤓

**Comment by DillFrescott** [comment]
@44vince44 Ok. I followed your instructions and it crashes precisely at 1:38:20 each time.

How do I get it to display the milliseconds?

**Comment by DillFrescott** [comment]
@Dendraspis its smooth as butter and just goes dark and ends.

**Comment by DillFrescott** [comment]
@Dendraspis If I sent you the original source file, would you like to try to encode it and see if the crash happens?

**Comment by Dendraspis** [comment]
You meant the encoded file, right?  But how does the source file behave on that position ?

**Comment by DillFrescott** [comment]
Yes, the encoded file crashes at that position.

The source file has no noticeable issues whatsoever at that time stamp (or anywhere in the movie at all)

**Comment by Dendraspis** [comment]
Seems to be a good decision before we start comment no.300. 🤔

**Comment by DillFrescott** [comment]
https://u.pcloud.link/publink/show?code=XZoaChXZRi7FNE0hBAz1ggooQVoXTzFT7w4y

Here ya go.

**Comment by 44vince44** [comment]
@DillFrescott I'll need something smaller, since 3.5GB is too much for me to handle:

Do you think you're able to select a range around the time index  1:38:20 (let's say approximately something like 1:37:00  to 1:39:00) and create an encode with video encoder set to copy/mux, and audio encoder set to copy/mux as well ?

The resulting file would be small and I'll be able to download it.

**Comment by Dendraspis** [comment]
I'm downloading @ 400kb/s ... will take a while.  😂 

@44vince44 
`ffmpeg` should be even easier. 
But 3.3GB too big? That makes me curious.  😲

**Comment by 44vince44** [comment]
@Dendraspis I'm in the third world ... not in Europe (although I lived there more than 25 years)

**Comment by DillFrescott** [comment]
@44vince44 I'll trim it and get it uploaded as soon as I can

**Comment by 44vince44** [comment]
@DillFrescott thank you. Please make sure you're not reencoding. and you're doing it copy/mux.

But later, you could do an aomenc encode of the trimmed range to see it crashes. That test would take minutes only.

**Comment by DillFrescott** [comment]
@44vince44 

https://u.pcloud.link/publink/show?code=XZ9LxhXZEtoivLTVNdfJffAAJOWOxkl6o1UV

**Comment by Dendraspis** [comment]
@44vince44 
So Egypt or South Africa I guess. 🤓

Yeah, would be interesting, if the 2min sample also fails.

If necessary I can create the sample...

**Comment by 44vince44** [comment]
Thanks I've got it, beginning some tests

**Comment by 44vince44** [comment]
@Dendraspis I'll give you a hint: we have ports that explode and corruption and many bad things ...

**Comment by DillFrescott** [comment]
It crashed on the trim version...

https://gist.github.com/DillFrescott/1c2ae4df95b85be6a5eeea90bbb7d756

**Comment by 44vince44** [comment]
I see, I'm doing some tests now, but my processor is only i7, slower...

**Comment by 44vince44** [comment]
Crashed also here ! Ok so now it's getting interesting

**Comment by DillFrescott** [comment]
Damn. I agree

**Comment by Dendraspis** [comment]
@44vince44 
Got it .... wow. 😮

**Comment by 44vince44** [comment]
@DillFrescott it seems the black frames just after 1min20 are not black frames, but they are resulting from a bad cut or something..
I'm continuing tests...

**Comment by DillFrescott** [comment]
Thats odd. Could that be the case with the other movie as well?

**Comment by 44vince44** [comment]
@DillFrescott it passed x265 encode, using ffmpeg pipe... So it's a problem with aomenc. Maybe it should be submitted to aomenc coders...

**Comment by DillFrescott** [comment]
I can try to submit something to them about it

**Comment by 44vince44** [comment]
I don't know who they are...

**Comment by DillFrescott** [comment]
Your guess is as good as mine. I just know where to find the newest build but thats about it.

https://ci.appveyor.com/project/marcomsousa/build-aom

**Comment by Dendraspis** [comment]
@44vince44 
Maybe you can try using AVS/VPY as decoder to see if it works better. 🤔

**Comment by 44vince44** [comment]
Wait I'm still doing tests...!!!

**Comment by DillFrescott** [comment]
https://github.com/marcomsousa/build_aom heres a github i found via that other link...

**Comment by 44vince44** [comment]
@Dendraspis just tried AVS pipe, it crashes!

**Comment by Dendraspis** [comment]
Is the x265 encode okay or is that section distorted ?! Maybe missing frames ?

**Comment by 44vince44** [comment]
x265 perfectly ok ... as Rav1e (the very slow one...)

**Comment by 44vince44** [comment]
Now, when I changed contrast and luma to avoid brutal transition, the bug still occurs.
What's happening here ? 
AVS => crash
VS=> crash
ffmpeg (dxva2) => crash 
Trying now qsvenc

**Comment by Dendraspis** [comment]
Did you use x265 with or without pipe?  😊

So it seems that aomenc itself is the problem, not the decoder or pipe. 

I guess there is a broken section, maybe just one bit, and x265 is able to handle it (accidentely).

**Comment by 44vince44** [comment]
@Dendraspis 

> Did you use x265 with or without pipe?😊

Both: I used direct AVS and FFMPEG pipe to reproduce same environment as with aomenc. They both were successful
Rav1e was successful also (with ffmpeg pipe)

Now with aomenc
AVS-> crash
VS-> crash
ffmpeg (dxva) -> crash
Qsvenc -> crash 

all crashes at the same frame.

**Comment by 44vince44** [comment]
aomenc in ONE PASS MODE does not crash. I think it's a problem with the stats file (it's a bug of aomenc)

**Comment by DillFrescott** [comment]
That actually makes a lot of sense considering it happened with three separate encodes and likely a lot more since I've tried probably about 6 so far and 3 failed.

**Comment by DillFrescott** [comment]
I'd prefer if one of you guys could take this issue up to the aomenc devs if we ever figure out how to contact them.

You would be able to explain it much better than I would.

**Comment by 44vince44** [comment]
Indeed, the only thing is to find the proper place to post aomenc issues...

**Comment by stax76** [comment]
It's here:

https://bugs.chromium.org/p/aomedia/issues

**Comment by 44vince44** [comment]
Thanks @stax76 this is most helpful!! I'm doing some more tests, I suspect it has something to do with duplicated frames...

**Comment by Dendraspis** [comment]
@44vince44 
Great idea with re-encoding the source file!  👍 
But I doubt we could/should address it to aomedia. It is definitely aomenc, that causes the crash, but we don't have the encoding settings nor we know how it was created nor we can reproduce it.  😞

**Comment by 44vince44** [comment]
@Dendraspis I agree, I have realized that later because the encode with rav1e did not fail... And beside it fails only in 2 pass mode with aomenc. I'm still doing some tests to understand and to see if there is a setting that allows to overcome that. I have some ideas and I'm trying them. I hope I'll find a solution...

**Comment by Dendraspis** [comment]
You already found a solution: use rav1e instead.  😁 

To be honest:
The only thing we know about the source is, that it's encoded with HEVC, probably using x265, and that ***mkvmerge v36.0.0*** was used. So what do you wanna find out? 🤔   Looks like a waste of even more time.

> WTF ? well at least there is no --bad switch (because it's probably default 🤣 )

🤣

**Comment by 44vince44** [comment]
LOOOOLL
 use rav1e instead. 😁

you should try it, you'd go crazy! rav1e has EVEN WORSE performance vs aomenc !!! 🤣

**Comment by Dendraspis** [comment]
I would leave it - 3.5GB for 118min Anime is absolutely okay in my opinion.

**Comment by 44vince44** [comment]
I'd do the same, and maybe re-encode it later when x266 is at its version 2 😁

**Comment by 44vince44** [comment]
@Dendraspis @DillFrescott I have a solution. I will not report that to them (which I can't locate still).

That's an exception that their team must conceptually solve... You need to place a keyframe, yet you can not.... I guess they'll find out about it...
SET **kf-max-dist=120**
I read somewhere you should set it to 5*fps
tried also 125, worked... I'm still not sure what kind or problem we're working around here 😕

**Comment by Dendraspis** [comment]
Congratulations!  🥳  🎉 

Who would have thought, that a default value for keyframes would cause such problems?!  😲

**Comment by 44vince44** [comment]
Thanks, indeed, who would have thought. This shows that their codec isn't really ready. It couldn't handle a violent scene change when last keyframe happened too long ago...

**Comment by 44vince44** [comment]
@DillFrescott are you following? we found the solution and fix

**Comment by DillFrescott** [comment]
Yup. Thanks guys I really appreciate it!
